src/test/groovy/com/jvm_bloggers/frontend/admin_area/AdminSocialChannelsPageSpec.groovy

```diff
@@ -0,0 +1,45 @@
+package com.jvm_bloggers.frontend.admin_area
+import com.jvm_bloggers.MockSpringContextAwareSpecification
+import org.apache.wicket.util.tester.FormTester
+
+class AdminSocialChannelsPageSpec extends MockSpringContextAwareSpecification {
+    static final String SAMPLE_VALID_LINK = "http://link.com"
+    static final String SAMPLE_VALID_MESSAGE = "valid message"
+
+    AdminSocialChannelsPageBackingBean backingBean = Mock(AdminSocialChannelsPageBackingBean)
+
+    @Override
+    protected void setupContext() {
+        addBean(backingBean)
+        addBean(Stub(PaginationConfiguration))
+    }
+
+    def "Should create new FacebookPost"() {
+        given:
+        currentUserIsAdmin()
+        tester.startPage(AdminSocialChannelsPage)
+
+        when:
+        FormTester formTester = tester.newFormTester(AdminSocialChannelsPage.FACEBOOK_POST_FORM_ID, false)
+        formTester.setValue(AdminSocialChannelsPage.LINK_INPUT_ID, SAMPLE_VALID_LINK)
+        formTester.setValue(AdminSocialChannelsPage.MESSAGE_INPUT_ID, SAMPLE_VALID_MESSAGE)
+        formTester.submit(AdminSocialChannelsPage.SAVE_BUTTON_ID)
+
+        then:
+        1 * backingBean.createNewFacebookPost(_,_)
+    }
+    def "Should not save FacebookPost when link is not valid"(){
```

Comment:
Assertion in this test does not check if the entity was not saved, only if backing bean publishing command was not called so the name is kind of misleading.